src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreTaskBase.cs

```diff
@@ -0,0 +1,291 @@
+                            CreateNoWindow = true,
+                            FileName = GetProcessFileName(ProcessFileName),
+                            RedirectStandardOutput = true,
+                            UseShellExecute = false,
+                            WorkingDirectory = Environment.CurrentDirectory,
+                        };
+
+                        // Place the output in the queue which handles logging messages coming through on StdOut
+                        process.OutputDataReceived += (sender, args) => loggingQueue.Enqueue(args?.Data);
+
+                        process.Exited += (sender, args) => semaphore.Release();
+
+                        try
+                        {
+                            Log.LogMessage(MessageImportance.Low, "Running command: \"{0}\" {1}", process.StartInfo.FileName, process.StartInfo.Arguments);
+
+                            process.Start();
+
+                            process.BeginOutputReadLine();
+
+                            semaphore.Wait(_cancellationTokenSource.Token);
+
+                            if (!process.HasExited)
+                            {
+                                try
+                                {
+                                    process.Kill();
+                                }
+                                catch (InvalidOperationException)
+                                {
+                                    // The process may have exited, in this case ignore the exception
+                                }
+                            }
+                        }
+                        catch (Exception e) when (
+                            e is OperationCanceledException
+                            || (e is AggregateException aggregateException && aggregateException.InnerException is OperationCanceledException))
+                        {
+                        }
+                    }
+                }
+                finally
+                {
+                    FileUtility.Delete(responseFile.FullName);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.LogErrorFromException(e);
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        /// <summary>
+        /// Gets the command-line arguments to use when launching the process that executes the restore.
+        /// </summary>
+        /// <returns>An <see cref="IEnumerable{String}" /> containing the command-line arguments that need to separated by spaces and surrounded by quotes.</returns>
+        internal virtual IEnumerable<string> GetCommandLineArguments(FileInfo responseFile)
+        {
+#if IS_CORECLR
+            // The full path to the executable for dotnet core
+            yield return Path.Combine(ThisAssemblyLazy.Value.DirectoryName, Path.ChangeExtension(ThisAssemblyLazy.Value.Name, ".Console.dll"));
```

Comment:
I guess it is intentional to not to use `responseFile` parameter when `IS_CORECLR` is true, correct?